File: backend/queues/src/main/java/com/griddynamics/msd365fp/manualreview/queues/model/dto/ItemDTO.java
package com.griddynamics.msd365fp.manualreview.queues.model.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.griddynamics.msd365fp.manualreview.model.*;
import com.griddynamics.msd365fp.manualreview.model.dfp.MainPurchase;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotNull;
import java.time.OffsetDateTime;
import java.util.Set;
import java.util.SortedSet;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ItemDTO {
    @NotNull
    private String id;
    @NotNull
    private OffsetDateTime imported;
    private OffsetDateTime updated;

    @NotNull
    private boolean active;

    private SortedSet<ItemNote> notes;
    private Set<String> tags;

    private MainPurchase purchase;
    private Decision decision;

    private ItemLock lock;
    private ItemEscalation escalation;
    private ItemHold hold;

    private Set<String> reviewers;
}

File: frontend/src/data-services/api-services/models/item-dto.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { ItemHoldDTO } from './item-hold-dto';
import { ItemLockDTO } from './item-lock-dto';
import { ItemNoteDTO } from './item-note-dto';
import { PurchaseDTO } from './purchase-dto';
import { DecisionDTO } from './decision-dto';
import { ItemLabelDTO } from './item-label-dto';

/**
 * Item details model from API
 */
export interface ItemDTO {
    id: string;
    /* string($date-time) */
    imported: string;
    /* string($date-time) */
    enriched: string;
    /* string($date-time) */
    active: boolean;
    label: ItemLabelDTO;
    notes: ItemNoteDTO[];
    tags: string[];
    purchase: PurchaseDTO;
    decision?: DecisionDTO;
    lock: ItemLockDTO;
    hold: ItemHoldDTO;
    reviewers: string[];
    ttl: number;
    get_etag: string;
}
